### PR TITLE
Joseph

### PR DESCRIPTION
**Pull request by ductran67**

### CODE REVIEW

**Comment by ductran67** [comment]
@jarkhyun 
@pablomarcel 
@nadenjohn 
Hi C-team, this is the big update for our project. I will send you guys the update info and the instruction how to work on it via slack. See you then.
Best,
Joseph

**Comment by cherimarie** [comment]
In general, we want to keep pull requests small. One of this scale is very challenging for team members to review. I'm also not seeing a reference to an Issue, so I don't know what goals this code is supposed to be accomplishing. In the future, remember it's fine to do multiple pull requests for an Issue if the work can be broken up into smaller chunks.

File: backend/controllers/myFavoritePostController.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        'foreignField': '_id', 
+        'as': 'post'
+      }
+    }, {
+      '$sort': {
+        'updatedAt': -1
+      }
+    }
+  ];
+  const myFavoritePosts = await MyFavoritePost.aggregate(pipeline);
```
Comment (cherimarie) [inline backend/controllers/myFavoritePostController.js:30]: Data interfacing should be separate from response/request handling. This logic should be in the model or data interface, in the pattern we've used for all our in-class projects.

Comment (cherimarie) [reply backend/controllers/myFavoritePostController.js:30]: Same comment applies throughout these Controller files.

File: backend/middleware/errorMiddleware.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const errorHandler = (err, req, res, next) => {
+  const statusCode = res.statusCode ? res.statusCode : 500
+
+  res.status(statusCode)
+
+  res.json({
+    message: err.message,
+    stack: process.env.NODE_ENV === 'production' ? null : err.stack,
```
Comment (cherimarie) [inline backend/middleware/errorMiddleware.js:8]: 💯

File: frontend/src/App.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  //   dispatch(getAllPosts())
+
+  //   return () => {
+  //     dispatch(reset())
+  //   }
+  // }, [isError, message, dispatch])
+
+  // if (isLoading) {
+  //   return <Spinner />
+  // }
```
Comment (cherimarie) [inline frontend/src/App.js:37]: Commented out code should not be merged. If you need to recover it after deletion, you can always look at the commit where it was removed to see it.

**Comment by cherimarie** [comment]
After looking through the code, this is far too large of a scope for a PR. There's no way for teammates to properly review. I strongly recommend breaking this work up into at least 4 PRs: 

- core backend (route handlers with no logic, to allow for review of path design)
- backend wired to data (handlers interacting with data interface methods, to allow for review of logic and code architecture)
- core frontend (base React app with no page content, wired to API, to allow for review of site design and API connection)
- frontend with content & interactivity (to allow for review of page content and logic)

To break this up, I'd start a new branch off of main and copy/paste in the code to accomplish the first bullet point above. After the code is reviewed and refactored, merge, then start a new branch off main and using this `joseph` branch as a model, write the code for the next set of work, and repeat.

File: backend/controllers/myFavoritePostController.js (hunk 1 above)
Comment (ductran67) [reply backend/controllers/myFavoritePostController.js:30]: Hi Cheri,
The controller files here play the same roles as data interface files that we learn in the class, and all the files in routes folder work as the same routes files from our class projects. I think it's just the naming issue.
Best,
Joseph

**Comment by pablomarcel** [comment]
@cherimarie 

Cheri, can we just merge this pull request and take the hit on points for not creating enough pull requests?
We understand the importance of making incremental pull request and that a massive code update is difficult to review. But the ship has sailed and I want to keep this project moving forward. 
There are still open items like implementing search, adding indexes, deploying and documenting. Going forward, those code updates will be handled in incremental pull requests. 
Going back and splitting the code and making pull requests will be time consuming and detrimental to the project. Yes, code is massive, but easy to understand and more importantly, it just works. I rather have working code than anything else. Other thing the pull request is massive because Joseph is an unstoppable creative force. I can't limit that.
Hope this pragmatic proposal makes sense.
Let us know.

**Comment by cherimarie** [comment]
@pablomarcel do what you need to do

**Pull request merged**